filebeat/input/filestream/environment_test.go

```diff
@@ -335,6 +333,39 @@ func (e *inputTestingEnvironment) waitUntilEventCount(count int) {
 	}
 }
 
+// waitUntilEventCountCtx calls waitUntilEventCount, but fails if ctx is cancelled.
+func (e *inputTestingEnvironment) waitUntilEventCountCtx(ctx context.Context, count int) {
+	e.t.Helper()
+	ch := make(chan struct{})
+
+	go func() {
+		e.waitUntilEventCount(count)
+		ch <- struct{}{}
+	}()
+
+	select {
+	case <-ctx.Done():
+		logLines := map[string][]string{}
+		for _, e := range e.pipeline.GetAllEvents() {
+			flat := e.Fields.Flatten()
+			pathi, _ := flat.GetValue("log.file.path")
+			path := pathi.(string)
+			msgi, _ := flat.GetValue("message")
+			msg := msgi.(string)
+			logLines[path] = append(logLines[path], msg)
+		}
+		ll := e.pipeline.GetAllEvents()
+		_ = ll
+		e.t.Fatalf("waitUntilEventCountCtx: %v. Want %d events, got %d: %v",
+			ctx.Err(),
+			count,
+			len(e.pipeline.GetAllEvents()),
+			logLines)
```

Comment:
Not a big deal, but we already get the line who generated the error in the test output, no need to add the method name here.